parser.go

```diff
@@ -143,9 +143,9 @@ func extractVarValue(src []byte, vars map[string]string) (value string, rest []b
 		}
 
 		// Work backwards to check if the line ends in whitespace then
-		// a comment (ie asdasd # some comment)
-		for i := endOfVar - 1; i >= 0; i-- {
-			if line[i] == charComment && i > 0 {
+		// a comment (ie asdasd # some comment # other)
```

Comment:
I would like to suggest this, because what you add increase the confusion

```suggestion
		// a comment (ie `whatever # some comment # other`)
```

Or any variation

```suggestion
		// a comment, ie: `whatever # some comment # other`
```

Or

```suggestion
		// a comment, ie: whatever # some comment # other
```

Because asdasd is unclear. I know the comment is not yours, and you are only moving it. But now, the confusion is increased.

I mean, I'm not even sure if `whatever` should be `whatever `, `foo=bar `

Reply:
I changed to another comment.